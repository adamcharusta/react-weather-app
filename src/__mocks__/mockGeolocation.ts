import { GeolocationType } from '../types/geolocationTypes';

export const mockGeolocation: GeolocationType = {
  id: 2553604,
  name: 'White House',
  latitude: 33.58831,
  longitude: -7.61138,
  elevation: 27,
  feature_code: 'PPLA',
  country_code: 'MA',
  admin1_id: 11281879,
  admin2_id: 6547294,
  timezone: 'Africa/Casablanca',
  population: 3144909,
  country_id: 2542007,
  country: 'Morocco',
  admin1: 'Casablanca-Settat',
  admin2: 'Casablanca',
};
