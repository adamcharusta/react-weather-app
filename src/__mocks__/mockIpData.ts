import { IpResponseType } from '../types/ipTypes';

export const mockIpData: IpResponseType = {
  ip: '45.138.153.39',
  network: '45.138.152.0/23',
  version: 'IPv4',
  city: 'Gmina Piątek',
  region: 'Łódź Voivodeship',
  region_code: '10',
  country: 'PL',
  country_name: 'Poland',
  country_code: 'PL',
  country_code_iso3: 'POL',
  country_capital: 'Warsaw',
  country_tld: '.pl',
  continent_code: 'EU',
  in_eu: true,
  postal: '99-120',
  latitude: 52.0696,
  longitude: 19.4734,
  timezone: 'Europe/Warsaw',
  utc_offset: '+0100',
  country_calling_code: '+48',
  currency: 'PLN',
  currency_name: 'Zloty',
  languages: 'pl',
  country_area: 312685,
  country_population: 37978548,
  asn: 'AS50411',
  org: 'ITV Media Sp. z o.o.',
};
