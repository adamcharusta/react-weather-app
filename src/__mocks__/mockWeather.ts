import moment from 'moment';
import { WeatherType } from '../types/weatherTypes';

export const mockWeather: WeatherType = {
  latitude: 33.5625,
  longitude: -7.625,
  generationtime_ms: 0.7530450820922852,
  utc_offset_seconds: 0,
  timezone: 'Africa/Casablanca',
  timezone_abbreviation: '+00',
  elevation: 23,
  current_weather: {
    temperature: 16.2,
    windspeed: 9.3,
    winddirection: 16,
    weathercode: 1,
    time: moment('2023-03-19T19:00').toDate(),
  },
  hourly_units: {
    time: 'iso8601',
    temperature_2m: 'Â°C',
    relativehumidity_2m: '%',
    windspeed_10m: 'km/h',
    rain: 'mm',
    snowfall: 'cm',
    visibility: 'm',
  },
  hourly: {
    time: [
      moment('2023-03-19T00:00').toDate(),
      moment('2023-03-19T01:00').toDate(),
      moment('2023-03-19T02:00').toDate(),
      moment('2023-03-19T03:00').toDate(),
      moment('2023-03-19T04:00').toDate(),
      moment('2023-03-19T05:00').toDate(),
      moment('2023-03-19T06:00').toDate(),
      moment('2023-03-19T07:00').toDate(),
      moment('2023-03-19T08:00').toDate(),
      moment('2023-03-19T09:00').toDate(),
      moment('2023-03-19T10:00').toDate(),
      moment('2023-03-19T11:00').toDate(),
      moment('2023-03-19T12:00').toDate(),
      moment('2023-03-19T13:00').toDate(),
      moment('2023-03-19T14:00').toDate(),
      moment('2023-03-19T15:00').toDate(),
      moment('2023-03-19T16:00').toDate(),
      moment('2023-03-19T17:00').toDate(),
      moment('2023-03-19T18:00').toDate(),
      moment('2023-03-19T19:00').toDate(),
      moment('2023-03-19T20:00').toDate(),
      moment('2023-03-19T21:00').toDate(),
      moment('2023-03-19T22:00').toDate(),
      moment('2023-03-19T23:00').toDate(),
      moment('2023-03-20T00:00').toDate(),
      moment('2023-03-20T01:00').toDate(),
      moment('2023-03-20T02:00').toDate(),
      moment('2023-03-20T03:00').toDate(),
      moment('2023-03-20T04:00').toDate(),
      moment('2023-03-20T05:00').toDate(),
      moment('2023-03-20T06:00').toDate(),
      moment('2023-03-20T07:00').toDate(),
      moment('2023-03-20T08:00').toDate(),
      moment('2023-03-20T09:00').toDate(),
      moment('2023-03-20T10:00').toDate(),
      moment('2023-03-20T11:00').toDate(),
      moment('2023-03-20T12:00').toDate(),
      moment('2023-03-20T13:00').toDate(),
      moment('2023-03-20T14:00').toDate(),
      moment('2023-03-20T15:00').toDate(),
      moment('2023-03-20T16:00').toDate(),
      moment('2023-03-20T17:00').toDate(),
      moment('2023-03-20T18:00').toDate(),
      moment('2023-03-20T19:00').toDate(),
      moment('2023-03-20T20:00').toDate(),
      moment('2023-03-20T21:00').toDate(),
      moment('2023-03-20T22:00').toDate(),
      moment('2023-03-20T23:00').toDate(),
      moment('2023-03-21T00:00').toDate(),
      moment('2023-03-21T01:00').toDate(),
      moment('2023-03-21T02:00').toDate(),
      moment('2023-03-21T03:00').toDate(),
      moment('2023-03-21T04:00').toDate(),
      moment('2023-03-21T05:00').toDate(),
      moment('2023-03-21T06:00').toDate(),
      moment('2023-03-21T07:00').toDate(),
      moment('2023-03-21T08:00').toDate(),
      moment('2023-03-21T09:00').toDate(),
      moment('2023-03-21T10:00').toDate(),
      moment('2023-03-21T11:00').toDate(),
      moment('2023-03-21T12:00').toDate(),
      moment('2023-03-21T13:00').toDate(),
      moment('2023-03-21T14:00').toDate(),
      moment('2023-03-21T15:00').toDate(),
      moment('2023-03-21T16:00').toDate(),
      moment('2023-03-21T17:00').toDate(),
      moment('2023-03-21T18:00').toDate(),
      moment('2023-03-21T19:00').toDate(),
      moment('2023-03-21T20:00').toDate(),
      moment('2023-03-21T21:00').toDate(),
      moment('2023-03-21T22:00').toDate(),
      moment('2023-03-21T23:00').toDate(),
      moment('2023-03-22T00:00').toDate(),
      moment('2023-03-22T01:00').toDate(),
      moment('2023-03-22T02:00').toDate(),
      moment('2023-03-22T03:00').toDate(),
      moment('2023-03-22T04:00').toDate(),
      moment('2023-03-22T05:00').toDate(),
      moment('2023-03-22T06:00').toDate(),
      moment('2023-03-22T07:00').toDate(),
      moment('2023-03-22T08:00').toDate(),
      moment('2023-03-22T09:00').toDate(),
      moment('2023-03-22T10:00').toDate(),
      moment('2023-03-22T11:00').toDate(),
      moment('2023-03-22T12:00').toDate(),
      moment('2023-03-22T13:00').toDate(),
      moment('2023-03-22T14:00').toDate(),
      moment('2023-03-22T15:00').toDate(),
      moment('2023-03-22T16:00').toDate(),
      moment('2023-03-22T17:00').toDate(),
      moment('2023-03-22T18:00').toDate(),
      moment('2023-03-22T19:00').toDate(),
      moment('2023-03-22T20:00').toDate(),
      moment('2023-03-22T21:00').toDate(),
      moment('2023-03-22T22:00').toDate(),
      moment('2023-03-22T23:00').toDate(),
      moment('2023-03-23T00:00').toDate(),
      moment('2023-03-23T01:00').toDate(),
      moment('2023-03-23T02:00').toDate(),
      moment('2023-03-23T03:00').toDate(),
      moment('2023-03-23T04:00').toDate(),
      moment('2023-03-23T05:00').toDate(),
      moment('2023-03-23T06:00').toDate(),
      moment('2023-03-23T07:00').toDate(),
      moment('2023-03-23T08:00').toDate(),
      moment('2023-03-23T09:00').toDate(),
      moment('2023-03-23T10:00').toDate(),
      moment('2023-03-23T11:00').toDate(),
      moment('2023-03-23T12:00').toDate(),
      moment('2023-03-23T13:00').toDate(),
      moment('2023-03-23T14:00').toDate(),
      moment('2023-03-23T15:00').toDate(),
      moment('2023-03-23T16:00').toDate(),
      moment('2023-03-23T17:00').toDate(),
      moment('2023-03-23T18:00').toDate(),
      moment('2023-03-23T19:00').toDate(),
      moment('2023-03-23T20:00').toDate(),
      moment('2023-03-23T21:00').toDate(),
      moment('2023-03-23T22:00').toDate(),
      moment('2023-03-23T23:00').toDate(),
      moment('2023-03-24T00:00').toDate(),
      moment('2023-03-24T01:00').toDate(),
      moment('2023-03-24T02:00').toDate(),
      moment('2023-03-24T03:00').toDate(),
      moment('2023-03-24T04:00').toDate(),
      moment('2023-03-24T05:00').toDate(),
      moment('2023-03-24T06:00').toDate(),
      moment('2023-03-24T07:00').toDate(),
      moment('2023-03-24T08:00').toDate(),
      moment('2023-03-24T09:00').toDate(),
      moment('2023-03-24T10:00').toDate(),
      moment('2023-03-24T11:00').toDate(),
      moment('2023-03-24T12:00').toDate(),
      moment('2023-03-24T13:00').toDate(),
      moment('2023-03-24T14:00').toDate(),
      moment('2023-03-24T15:00').toDate(),
      moment('2023-03-24T16:00').toDate(),
      moment('2023-03-24T17:00').toDate(),
      moment('2023-03-24T18:00').toDate(),
      moment('2023-03-24T19:00').toDate(),
      moment('2023-03-24T20:00').toDate(),
      moment('2023-03-24T21:00').toDate(),
      moment('2023-03-24T22:00').toDate(),
      moment('2023-03-24T23:00').toDate(),
      moment('2023-03-25T00:00').toDate(),
      moment('2023-03-25T01:00').toDate(),
      moment('2023-03-25T02:00').toDate(),
      moment('2023-03-25T03:00').toDate(),
      moment('2023-03-25T04:00').toDate(),
      moment('2023-03-25T05:00').toDate(),
      moment('2023-03-25T06:00').toDate(),
      moment('2023-03-25T07:00').toDate(),
      moment('2023-03-25T08:00').toDate(),
      moment('2023-03-25T09:00').toDate(),
      moment('2023-03-25T10:00').toDate(),
      moment('2023-03-25T11:00').toDate(),
      moment('2023-03-25T12:00').toDate(),
      moment('2023-03-25T13:00').toDate(),
      moment('2023-03-25T14:00').toDate(),
      moment('2023-03-25T15:00').toDate(),
      moment('2023-03-25T16:00').toDate(),
      moment('2023-03-25T17:00').toDate(),
      moment('2023-03-25T18:00').toDate(),
      moment('2023-03-25T19:00').toDate(),
      moment('2023-03-25T20:00').toDate(),
      moment('2023-03-25T21:00').toDate(),
      moment('2023-03-25T22:00').toDate(),
      moment('2023-03-25T23:00').toDate(),
    ],
    temperature_2m: [
      14.7, 14.5, 14.5, 14.3, 13.9, 13.7, 13, 12.7, 13.4, 15, 16.7, 18, 18.5, 18.9, 18.9, 18.8,
      18.4, 17.8, 17, 16.2, 15.9, 15.7, 15.5, 15.3, 14.8, 14.2, 13.9, 13.7, 13.4, 13.2, 13, 13,
      14.3, 16.6, 18.3, 19.4, 20.1, 20.7, 21.2, 21.1, 20.4, 19.5, 18.3, 17.4, 17, 16.6, 16.2, 15.8,
      15.3, 15, 14, 12.8, 12, 11.7, 11.6, 11.5, 13, 15.8, 18, 19.4, 20.2, 20.6, 20.6, 20.3, 19.5,
      18.7, 17.9, 16.9, 16, 15.6, 15, 14.8, 14.7, 14.6, 14.1, 13.8, 13.3, 12.9, 12.1, 11.5, 12.6,
      16, 17.9, 19.3, 20.7, 21.2, 20.8, 20.7, 20.2, 19.2, 18.2, 17.4, 16.7, 15.9, 15.6, 15.3, 15,
      14.6, 14.1, 13.5, 13.2, 12.8, 12.8, 13.1, 13.7, 14.9, 16.5, 18.5, 20.5, 20.8, 20.6, 20.1,
      19.7, 19.2, 18.4, 17.6, 16.8, 15.9, 15.6, 15.4, 15.1, 14.9, 14.6, 14.2, 13.8, 13.2, 13.1,
      13.6, 14.6, 16.1, 17.4, 18.9, 20.4, 19.9, 19.9, 19.5, 19.2, 18.6, 17.9, 17.3, 16.6, 15.9,
      15.6, 15.5, 15.3, 15.2, 15.2, 15.1, 14.9, 14.8, 14.8, 15.2, 15.7, 16.6, 17.4, 18.3, 19.2,
      19.6, 19.9, 19.9, 19.7, 19.3, 18.7, 18.1, 17.3, 16.6, 16.3, 16.2,
    ],
    relativehumidity_2m: [
      83, 86, 86, 85, 88, 90, 93, 92, 84, 73, 65, 60, 62, 58, 59, 60, 64, 65, 72, 78, 80, 81, 82,
      85, 90, 92, 91, 89, 88, 86, 84, 84, 73, 52, 42, 42, 43, 45, 47, 49, 51, 52, 63, 67, 69, 73,
      77, 82, 86, 85, 91, 94, 94, 93, 90, 87, 78, 67, 69, 60, 59, 61, 62, 61, 67, 71, 74, 79, 87,
      92, 95, 95, 91, 86, 89, 93, 93, 93, 95, 96, 89, 76, 68, 60, 51, 47, 53, 53, 59, 72, 80, 86,
      90, 94, 95, 94, 94, 95, 97, 99, 100, 100, 100, 96, 91, 82, 74, 64, 55, 54, 57, 61, 64, 67, 73,
      78, 84, 91, 94, 95, 96, 96, 94, 93, 94, 96, 95, 90, 82, 73, 69, 66, 63, 68, 67, 67, 69, 72,
      77, 82, 87, 92, 93, 93, 93, 94, 94, 95, 95, 95, 94, 91, 87, 82, 78, 74, 70, 67, 64, 63, 66,
      71, 78, 83, 89, 94, 96, 96,
    ],
    windspeed_10m: [
      3.8, 3.8, 3.6, 4.7, 5.2, 5, 4.3, 4.1, 3.8, 4.6, 3.7, 7.8, 10.4, 12.5, 14.4, 15.1, 15.1, 14.4,
      12.3, 9.3, 9.4, 8.2, 6.9, 6.7, 6.2, 5.1, 4.3, 4.2, 4.3, 4.5, 4.9, 4.8, 5.9, 8.9, 11.5, 12.3,
      13.8, 14.3, 14.4, 14.8, 14.7, 15.2, 13, 9.5, 7.2, 5.6, 5.4, 4.2, 3.1, 3.3, 4, 4, 4.4, 4.3, 4,
      4.6, 2.9, 2.5, 3.5, 7.2, 8.3, 9.4, 10.1, 10.7, 10.6, 8.5, 6.2, 4, 2.6, 2.9, 2.7, 1.8, 2.6,
      3.5, 3.7, 4.2, 3.6, 3.5, 3.8, 3.7, 3.3, 1.1, 4.7, 6.7, 6.8, 9.7, 10.5, 11.2, 10.5, 10.8, 8.2,
      6.6, 4.9, 3.7, 3.2, 2.9, 2.3, 1.8, 2.4, 3.3, 3.4, 3.2, 3.2, 3.4, 3.6, 3.5, 3.4, 5.2, 8.3, 10,
      11, 11.7, 11.5, 10.4, 9.2, 7.7, 6.4, 5, 4.8, 4.8, 4.8, 4.2, 3.6, 2.8, 2.2, 2.6, 3.6, 2.9, 3,
      4.8, 6.5, 8.2, 9.7, 15.5, 17.3, 18.7, 18.6, 17.9, 16.2, 13.8, 11.5, 9, 8.1, 7.6, 7.8, 7.5,
      7.7, 7.7, 7.7, 8, 8.6, 8.7, 9.3, 9.7, 10.7, 13.1, 16.2, 17.5, 18.1, 17.9, 16.6, 14.8, 12.1,
      10.1, 8.2, 6.1, 5.1, 4.3,
    ],
    rain: [
      0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
      0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
      0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
      0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
      0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
      0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
    ],
    snowfall: [
      0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
      0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
      0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
      0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
      0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
      0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
    ],
    visibility: [
      24140, 24140, 24140, 24140, 24140, 24140, 24140, 24140, 24140, 24140, 24140, 24140, 24140,
      24140, 24140, 24140, 24140, 24140, 24140, 24140, 24140, 24140, 24140, 24140, 24140, 24140,
      24140, 24140, 24140, 24140, 24140, 24140, 24140, 24140, 24140, 24140, 24140, 24140, 24140,
      24140, 24140, 24140, 24140, 24140, 24140, 24140, 24140, 24140, 24140, 24140, 24140, 24140,
      24140, 24140, 24140, 24140, 24140, 24140, 24140, 24140, 24140, 24140, 24140, 24140, 24140,
      24140, 24140, 24140, 24140, 24140, 24140, 24140, 24140, 24140, 24140, 24140, 24140, 24140,
      24140, 24140, 24140, 24140, 24140, 24140, 24140, 24140, 24140, 24140, 24140, 24140, 24140,
      24140, 24140, 24140, 24140, 24140, 24140, 24140, 24140, 24140, 24140, 24140, 24140, 24140,
      24140, 24140, 24140, 24140, 24140, 24140, 24140, 24140, 24140, 24140, 24140, 24140, 24140,
      24140, 24140, 24140, 24140, 24140, 24140, 24140, 24140, 24140, 24140, 24140, 24140, 24140,
      24140, 24140, 24140, 24140, 24140, 24140, 24140, 24140, 24140, 24140, 24140, 24140, 24140,
      24140, 24140, 24140, 24140, 24140, 24140, 24140, 24140, 24140, 24140, 24140, 24140, 24140,
      24140, 24140, 24140, 24140, 24140, 24140, 24140, 24140, 24140, 24140, 24140, 24140,
    ],
  },
};
