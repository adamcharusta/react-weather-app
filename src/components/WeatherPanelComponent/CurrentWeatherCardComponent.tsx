import React, { useContext } from 'react';
import AppContext from '../../appContext';
import {
  Box,
  Card,
  CardContent,
  Grid,
  Tooltip,
  Typography,
  useMediaQuery,
  useTheme,
} from '@mui/material';
import moment from 'moment-timezone';
import ArrowUpwardIcon from '@mui/icons-material/ArrowUpward';
import getWeatherDescription from './getWeatherDescription';
import AdministrationStackComponent from '../AdministrationStackComponent';
import Flag from 'react-world-flags';
import { TimeFormatEnum } from '../../types/timeFormatEnum';

const CurrentWeatherCardComponent = () => {
  const { geolocation, weather } = useContext(AppContext);

  const theme = useTheme();
  const isMediumScreen = useMediaQuery(theme.breakpoints.down('md'));

  return (
    <Card sx={{ marginTop: 2 }}>
      <CardContent>
        <Grid item>
          <Typography color='text.secondary' gutterBottom>
            {moment().format(TimeFormatEnum.DATE_TIME)}
          </Typography>
        </Grid>

        <Grid container justifyContent='space-between'>
          <Grid item xs={12} sm={12} md={6}>
            <Typography variant='h2'>
              <Flag code={geolocation?.country_code} style={{ width: 70 }} />
              {geolocation?.name}{' '}
            </Typography>
            <AdministrationStackComponent geolocation={geolocation} />
          </Grid>
          <Grid item container xs={12} sm={12} md={6}>
            <Box
              display='flex'
              justifyContent={isMediumScreen ? 'flex-start' : 'flex-end'}
              width='100%'
            >
              <Typography variant='h2'>
                {weather?.current_weather.temperature}
                {weather?.hourly_units.temperature_2m}
              </Typography>
            </Box>
          </Grid>
        </Grid>

        <Grid container>
          <Grid item xs={12} sm={12} md={6}>
            <Typography variant='h6'>
              <Tooltip title='Time of last measurement' arrow disableInteractive placement='bottom'>
                <span>
                  Time: {moment(weather?.current_weather.time).format(TimeFormatEnum.TIME)}{' '}
                  {weather?.timezone_abbreviation}
                </span>
              </Tooltip>
            </Typography>
          </Grid>

          <Grid item xs={12} sm={12} md={6}>
            <Typography variant='h6'>
              Local time:{' '}
              {moment()
                .tz(geolocation?.timezone as string)
                ?.format(TimeFormatEnum.TIME)}
            </Typography>
          </Grid>

          <Grid item xs={12} sm={12} md={6}>
            <Typography variant='h6'>
              Wind speed: {weather?.current_weather.windspeed} {weather?.hourly_units.windspeed_10m}
            </Typography>
          </Grid>

          <Grid item container xs={12} sm={12} md={6}>
            <Grid item>
              <Typography variant='h6'>
                Wind direction: {weather?.current_weather.winddirection}
              </Typography>
            </Grid>
            <Grid item>
              <ArrowUpwardIcon
                sx={{ transform: `rotate(${weather?.current_weather.winddirection}deg)` }}
              />
            </Grid>
          </Grid>

          <Grid item xs={12}>
            <Typography variant='h6'>
              <Tooltip
                title={`WMO Weather interpretation code: ${weather?.current_weather.weathercode}`}
                arrow
                disableInteractive
                placement='bottom'
              >
                <span>Weather: {getWeatherDescription(weather?.current_weather.weathercode)}</span>
              </Tooltip>
            </Typography>
          </Grid>
        </Grid>
      </CardContent>
    </Card>
  );
};

export default CurrentWeatherCardComponent;
