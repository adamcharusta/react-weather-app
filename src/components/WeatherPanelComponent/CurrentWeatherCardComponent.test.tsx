import React from 'react';
import { render } from '@testing-library/react';
import { mockGeolocation } from '../../__mocks__/mockGeolocation';
import AppContext from '../../appContext';
import { mockWeather } from '../../__mocks__/mockWeather';
import CurrentWeatherCardComponent from './CurrentWeatherCardComponent';

describe('CurrentWeatherCardComponent', () => {
  it('snapshot', () => {
    const component = render(
      <AppContext.Provider
        value={{
          geolocation: mockGeolocation,
          setGeolocation: jest.fn(),
          weather: mockWeather,
          setWeather: jest.fn(),
        }}
      >
        <CurrentWeatherCardComponent />
      </AppContext.Provider>,
    );

    expect(component.asFragment()).toMatchSnapshot();
  });
});
