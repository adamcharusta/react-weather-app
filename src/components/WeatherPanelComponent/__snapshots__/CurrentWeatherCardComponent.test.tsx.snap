// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrentWeatherCardComponent snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-wmn96w-MuiPaper-root-MuiCard-root"
  >
    <div
      class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-1ydyjjr-MuiTypography-root"
        >
          01-01-1970 01:00
        </p>
      </div>
      <div
        class="MuiGrid-root MuiGrid-container css-1xzqqmo-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-6 css-1fyyp8j-MuiGrid-root"
        >
          <h2
            class="MuiTypography-root MuiTypography-h2 css-1sra7t5-MuiTypography-root"
          >
            <img
              src="data:image/svg+xml,%3csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 900 600'%3e%3cpath transform='scale(-1 1)' style='fill:%23c1272d;fill-opacity:1;stroke:none' d='M-900 0H0v600h-900z'/%3e%3cpath d='m450 224.315-44.467 136.87 116.401-84.559h-143.87l116.403 84.559z' style='fill:none;stroke:%23006233;stroke-width:14.62993431;stroke-miterlimit:4;stroke-dasharray:none'/%3e%3c/svg%3e"
              style="width: 70px;"
            />
            White House 
          </h2>
          <div
            class="MuiStack-root css-w4z10b-MuiStack-root"
          >
            <span
              class="MuiTypography-root MuiTypography-body1 css-1pnmrwp-MuiTypography-root"
            >
              Casablanca
            </span>
            <hr
              class="MuiDivider-root MuiDivider-fullWidth MuiDivider-vertical MuiDivider-flexItem css-1tk1bks-MuiDivider-root"
            />
            <span
              class="MuiTypography-root MuiTypography-body1 css-1pnmrwp-MuiTypography-root"
            >
              Casablanca-Settat
            </span>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-6 css-18kb45e-MuiGrid-root"
        >
          <div
            class="MuiBox-root css-asvefe"
          >
            <h2
              class="MuiTypography-root MuiTypography-h2 css-1sra7t5-MuiTypography-root"
            >
              16.2Â°C
            </h2>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-6 css-1fyyp8j-MuiGrid-root"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
          >
            <span
              aria-label="Time of last measurement"
              class=""
              data-mui-internal-clone-element="true"
            >
              Time: 19:00 +00 
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-colorInfo MuiSvgIcon-fontSizeMedium css-m5zvh1-MuiSvgIcon-root"
                data-testid="InfoOutlinedIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M11 7h2v2h-2zm0 4h2v6h-2zm1-9C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
                />
              </svg>
            </span>
          </h6>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-6 css-1fyyp8j-MuiGrid-root"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
          >
            Local time: 00:00
          </h6>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-6 css-1fyyp8j-MuiGrid-root"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
          >
            Wind speed: 9.3 km/h
          </h6>
        </div>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-6 css-18kb45e-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <h6
              class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
            >
              Wind direction: 16
            </h6>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1wlw7z0-MuiSvgIcon-root"
              data-testid="ArrowUpwardIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="m4 12 1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8-8 8z"
              />
            </svg>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
          >
            <span
              aria-label="WMO Weather interpretation code: 1"
              class=""
              data-mui-internal-clone-element="true"
            >
              Weather: Mainly clear, partly cloudy, and overcast 
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-colorInfo MuiSvgIcon-fontSizeMedium css-m5zvh1-MuiSvgIcon-root"
                data-testid="InfoOutlinedIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M11 7h2v2h-2zm0 4h2v6h-2zm1-9C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
                />
              </svg>
            </span>
          </h6>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
